libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationTest.java

```diff
@@ -28,199 +28,257 @@
 
 public class MapboxNavigationTest extends BaseTest {
 
-  private MapboxNavigation navigation;
+  @Test
+  public void sanityTest() {
+    MapboxNavigation navigation = buildMapboxNavigation();
 
-  @Before
-  public void before() throws Exception {
-    navigation = new MapboxNavigation(mock(Context.class), ACCESS_TOKEN, mock(NavigationTelemetry.class),
-      mock(LocationEngine.class));
+    assertNotNull(navigation);
   }
 
   @Test
-  public void sanityTest() {
-    assertNotNull("should not be null", navigation);
+  public void sanityTestWithOptions() {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
-    assertNotNull("should not be null", navigationWithOptions);
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+
+    assertNotNull(navigationWithOptions);
   }
 
   @Test
   public void voiceMilestone_onInitializationDoesGetAdded() throws Exception {
-    assertTrue(navigation.getMilestones().get(0).getIdentifier() == VOICE_INSTRUCTION_MILESTONE_ID);
+    MapboxNavigation navigation = buildMapboxNavigation();
+
+    int identifier = navigation.getMilestones().get(0).getIdentifier();
+
+    assertEquals(identifier, VOICE_INSTRUCTION_MILESTONE_ID);
   }
 
   @Test
   public void bannerMilestone_onInitializationDoesGetAdded() throws Exception {
-    assertTrue(navigation.getMilestones().get(1).getIdentifier() == BANNER_INSTRUCTION_MILESTONE_ID);
+    MapboxNavigation navigation = buildMapboxNavigation();
+
+    int identifier = navigation.getMilestones().get(1).getIdentifier();
+
+    assertEquals(identifier, BANNER_INSTRUCTION_MILESTONE_ID);
   }
 
 
   @Test
   public void defaultMilestones_onInitializationDoNotGetAdded() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().defaultMilestonesEnabled(false).build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+
     assertEquals(0, navigationWithOptions.getMilestones().size());
```

Comment:
👀 https://github.com/mapbox/mapbox-navigation-android/pull/866#discussion_r182155984
`navigationWithOptions.getMilestones().size()` is the `Act` here.
This applies to other tests included in `MapboxNavigationTest` 👇